openmdao/drivers/test/test_pyoptsparse.py

```diff
@@ -1191,5 +1191,38 @@ def solve_nonlinear(self, params, unknowns, resids):
         sub_sparsity = prob.driver.sub_sparsity
         self.assertEquals(len(sub_sparsity['seg0.r_i']['y_i']), 9)
 
+
+    def test_snopt_diff(self):
+        if OPTIMIZER is not 'SNOPT':
+            raise unittest.SkipTest()
+
+        prob = Problem()
+        root = prob.root = Group()
+
+        root.add('p1', IndepVarComp('x', 50.0), promotes=['*'])
+        root.add('p2', IndepVarComp('y', 50.0), promotes=['*'])
+        root.add('comp', Paraboloid(), promotes=['*'])
+        root.add('con', ExecComp('c = - x + y'), promotes=['*'])
+
+        prob.driver = pyOptSparseDriver()
+        prob.driver.options['optimizer'] = OPTIMIZER
+        prob.driver.options['snopt_diff'] = True
+        
+        prob.driver.options['print_results'] = False
+        prob.driver.add_desvar('x', lower=-50.0, upper=50.0)
+        prob.driver.add_desvar('y', lower=-50.0, upper=50.0)
+
+        prob.driver.add_objective('f_xy')
+        prob.driver.add_constraint('c', upper=-15.0)
+
+        prob.setup(check=False)
+        prob.run()
+
+        # Minimum should be at (7.166667, -7.833334)
+        assert_rel_error(self, prob['x'], 7.16667, 1e-6)
+        assert_rel_error(self, prob['y'], -7.833334, 1e-6)
+
+        
```

Comment:
One thing you could do to be absolutely certain that the openmdao derivatives aren't still being called (and to test it) is to create a new class that inherits from `Paraboloid`, and then give it an `apply_linear` function that does nothing except raise an exception if called. The apply_linear function would get called if openmdao took a derivative.